### ISSUE TITLE
Truffle deployProxy() Error: Unsuported TypeName node type: Mapping

### ISSUE THREAD

**Issue by rok4f** [OpenZeppelin/openzeppelin-upgrades#365]
Hi all:

See sample project at [deployproxy-test](https://github.com/rok4f/deployproxy-test) .

Migration code:

```js
const { deployProxy } = require('@openzeppelin/truffle-upgrades');
const NestGovernance = artifacts.require("NestGovernance");

module.exports = async function(deployer) {
  const proxy = await deployProxy(NestGovernance, ['0x0000000000000000000000000000000000000000', '0x0000000000000000000000000000000000000000'], { deployer, initializer: 'initialize' });
  console.log("Deployed", proxy.address);
};
```

Run command:

```
npx truffle migrate
```

get error:

```
Compiling your contracts...
===========================
> Compiling ./contracts/Migrations.sol
> Compiling ./contracts/NestBase.sol
> Compiling ./contracts/NestGovernance.sol
> Compiling ./contracts/NestMapping.sol
> Compiling ./contracts/interface/INNIncome.sol
> Compiling ./contracts/interface/INToken.sol
> Compiling ./contracts/interface/INTokenController.sol
> Compiling ./contracts/interface/INestGovernance.sol
> Compiling ./contracts/interface/INestLedger.sol
> Compiling ./contracts/interface/INestMapping.sol
> Compiling ./contracts/interface/INestMining.sol
> Compiling ./contracts/interface/INestPriceFacade.sol
> Compiling ./contracts/interface/INestQuery.sol
> Compiling ./contracts/interface/INestRedeeming.sol
> Compiling ./contracts/interface/INestVote.sol
> Compiling ./contracts/interface/INest_NToken.sol
> Compiling ./contracts/interface/IProxyAdmin.sol
> Compiling ./contracts/interface/IVotePropose.sol
> Compiling ./contracts/lib/Context.sol
> Compiling ./contracts/lib/ERC20.sol
> Compiling ./contracts/lib/IERC20.sol
> Compiling ./contracts/lib/TransferHelper.sol
> Compiling ./contracts/test/IBNEST.sol
> Compiling ./contracts/test/ITransferable.sol
> Compiling ./contracts/test/NestMining2.sol
> Compiling ./contracts/test/PostInOneBlock.sol
> Compiling ./contracts/test/ProxyAdminTest.sol
> Compiling ./contracts/test/SetQueryPrice.sol
> Compiling ./contracts/test/SuperMan.sol
> Compiling ./contracts/test/TestERC20.sol
> Compiling ./contracts/test/UpdateAdmin.sol
> Compiling ./contracts/test/UpdateProxyPropose.sol
> Artifacts written to /Users/ninnin/deployproxy-test/build/contracts
> Compiled successfully using:
   - solc: 0.8.3+commit.8d00100c.Emscripten.clang

Starting migrations...
======================
> Network name:    'development'
> Network id:      1621633688592
> Block gas limit: 6721975 (0x6691b7)


1_initial_migration.js
======================

   Deploying 'Migrations'
   ----------------------
   > transaction hash:    0x2a31770eb66bd0e19ac3e5a163097bd2c1a388e2823727e96e76470a0dfb2529
   > Blocks: 0            Seconds: 0
   > contract address:    0x9F707326c208aE9F985bb4e4B544c13d20f7c498
   > block number:        7
   > block timestamp:     1621638230
   > account:             0x304e8ff7d3b09c37a24a9f9DC07e54cC4a73c935
   > balance:             99.9505249
   > gas used:            153718 (0x25876)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00307436 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.00307436 ETH


2_deploy_contracts.js
=====================

Error: Unsuported TypeName node type: Mapping
    at serializeTypeName (/Users/ninnin/node_modules/_@openzeppelin_upgrades-core@1.7.3@@openzeppelin/upgrades-core/src/utils/function.ts:37:13)
    at /Users/ninnin/node_modules/_@openzeppelin_upgrades-core@1.7.3@@openzeppelin/upgrades-core/src/utils/function.ts:22:62
    at Array.map (<anonymous>)
    at serializeTypeName (/Users/ninnin/node_modules/_@openzeppelin_upgrades-core@1.7.3@@openzeppelin/upgrades-core/src/utils/function.ts:22:48)
    at /Users/ninnin/node_modules/_@openzeppelin_upgrades-core@1.7.3@@openzeppelin/upgrades-core/src/utils/function.ts:43:23
    at Array.map (<anonymous>)
    at Object.getFunctionSignature (/Users/ninnin/node_modules/_@openzeppelin_upgrades-core@1.7.3@@openzeppelin/upgrades-core/src/utils/function.ts:43:6)
    at /Users/ninnin/node_modules/_@openzeppelin_upgrades-core@1.7.3@@openzeppelin/upgrades-core/src/validate/run.ts:168:25
    at Array.map (<anonymous>)
    at Object.validate (/Users/ninnin/node_modules/_@openzeppelin_upgrades-core@1.7.3@@openzeppelin/upgrades-core/src/validate/run.ts:168:12)
    at Object.validateArtifacts (/Users/ninnin/node_modules/_@openzeppelin_truffle-upgrades@1.7.0@@openzeppelin/truffle-upgrades/src/utils/validations.ts:19:10)
    at Object.deployImpl (/Users/ninnin/node_modules/_@openzeppelin_truffle-upgrades@1.7.0@@openzeppelin/truffle-upgrades/src/utils/deploy-impl.ts:29:23)
    at deployProxy (/Users/ninnin/node_modules/_@openzeppelin_truffle-upgrades@1.7.0@@openzeppelin/truffle-upgrades/src/deploy-proxy.ts:46:16)
    at module.exports (/Users/ninnin/deployproxy-test/migrations/2_deploy_contracts.js:11:17)
    at Migration._deploy (/home/ob1/.config/yarn/global/node_modules/truffle/build/webpack:/packages/migrate/Migration.js:79:1)
    at Migration._load (/home/ob1/.config/yarn/global/node_modules/truffle/build/webpack:/packages/migrate/Migration.js:61:1)
    at Migration.run (/home/ob1/.config/yarn/global/node_modules/truffle/build/webpack:/packages/migrate/Migration.js:212:1)
    at Object.runMigrations (/home/ob1/.config/yarn/global/node_modules/truffle/build/webpack:/packages/migrate/index.js:150:1)
    at Object.runFrom (/home/ob1/.config/yarn/global/node_modules/truffle/build/webpack:/packages/migrate/index.js:110:1)
    at Object.run (/home/ob1/.config/yarn/global/node_modules/truffle/build/webpack:/packages/migrate/index.js:87:1)
    at runMigrations (/home/ob1/.config/yarn/global/node_modules/truffle/build/webpack:/packages/core/lib/commands/migrate.js:258:1)
    at Object.run (/home/ob1/.config/yarn/global/node_modules/truffle/build/webpack:/packages/core/lib/commands/migrate.js:223:1)
    at Command.run (/home/ob1/.config/yarn/global/node_modules/truffle/build/webpack:/packages/core/lib/command.js:147:1)
Truffle v5.3.6 (core: 5.3.6)
Node v14.15.5
```

But after deleting `contracts/test/IBNEST.sol` it will be fine:

```
Compiling your contracts...
===========================
> Compiling ./contracts/Migrations.sol
> Compiling ./contracts/NestBase.sol
> Compiling ./contracts/NestGovernance.sol
> Compiling ./contracts/NestMapping.sol
> Compiling ./contracts/interface/INNIncome.sol
> Compiling ./contracts/interface/INToken.sol
> Compiling ./contracts/interface/INTokenController.sol
> Compiling ./contracts/interface/INestGovernance.sol
> Compiling ./contracts/interface/INestLedger.sol
> Compiling ./contracts/interface/INestMapping.sol
> Compiling ./contracts/interface/INestMining.sol
> Compiling ./contracts/interface/INestPriceFacade.sol
> Compiling ./contracts/interface/INestQuery.sol
> Compiling ./contracts/interface/INestRedeeming.sol
> Compiling ./contracts/interface/INestVote.sol
> Compiling ./contracts/interface/INest_NToken.sol
> Compiling ./contracts/interface/IProxyAdmin.sol
> Compiling ./contracts/interface/IVotePropose.sol
> Compiling ./contracts/lib/Context.sol
> Compiling ./contracts/lib/ERC20.sol
> Compiling ./contracts/lib/IERC20.sol
> Compiling ./contracts/lib/TransferHelper.sol
> Compiling ./contracts/test/ITransferable.sol
> Compiling ./contracts/test/NestMining2.sol
> Compiling ./contracts/test/PostInOneBlock.sol
> Compiling ./contracts/test/ProxyAdminTest.sol
> Compiling ./contracts/test/SetQueryPrice.sol
> Compiling ./contracts/test/SuperMan.sol
> Compiling ./contracts/test/TestERC20.sol
> Compiling ./contracts/test/UpdateAdmin.sol
> Compiling ./contracts/test/UpdateProxyPropose.sol
> Artifacts written to /Users/ninnin/deployproxy-test/build/contracts
> Compiled successfully using:
   - solc: 0.8.3+commit.8d00100c.Emscripten.clang



Starting migrations...
======================
> Network name:    'development'
> Network id:      1621633688592
> Block gas limit: 6721975 (0x6691b7)

1_initial_migration.js
======================

   Deploying 'Migrations'
   ----------------------
   > transaction hash:    0x4aa37ae555aec54ed6e8f59adf4d4260536bbbc495dbcd5ad56241240d3fd0e7
   > Blocks: 0            Seconds: 0
   > contract address:    0x1A246c700a56AD4Abc7b19DD063960a496fF5560
   > block number:        9
   > block timestamp:     1621639038
   > account:             0x304e8ff7d3b09c37a24a9f9DC07e54cC4a73c935
   > balance:             99.94660474
   > gas used:            153718 (0x25876)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00307436 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.00307436 ETH


2_deploy_contracts.js
=====================

   Deploying 'NestGovernance'
   --------------------------
   > transaction hash:    0xf62a2ce1a0f5d529644381fb6a98ac8470515abfdabe96681cab5fb607d15184
   > Blocks: 0            Seconds: 0
   > contract address:    0x40626aA12C6001802d1322e670Ce945a97D0bE45
   > block number:        11
   > block timestamp:     1621639039
   > account:             0x304e8ff7d3b09c37a24a9f9DC07e54cC4a73c935
   > balance:             99.92621994
   > gas used:            976950 (0xee836)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.019539 ETH


   Deploying 'ProxyAdmin'
   ----------------------
   > transaction hash:    0xd001370545b6333dd6e34677ddbfcec2b6bb104bc342b6eb7a36d8313123200d
   > Blocks: 0            Seconds: 0
   > contract address:    0x5252c8039A3CaF1597A9aEB93E13FcaB7245Ac97
   > block number:        12
   > block timestamp:     1621639039
   > account:             0x304e8ff7d3b09c37a24a9f9DC07e54cC4a73c935
   > balance:             99.91656554
   > gas used:            482720 (0x75da0)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.0096544 ETH


   Deploying 'TransparentUpgradeableProxy'
   ---------------------------------------
   > transaction hash:    0x4b09e6458d41457bbc7e792ba52cdd2188db66281d273835d8f3800060ab1aca
   > Blocks: 0            Seconds: 0
   > contract address:    0x51e165AB3DcFBaE0FC45F3c0685db4ec52ff730e
   > block number:        13
   > block timestamp:     1621639039
   > account:             0x304e8ff7d3b09c37a24a9f9DC07e54cC4a73c935
   > balance:             99.90382416
   > gas used:            637069 (0x9b88d)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.01274138 ETH

Deployed 0x51e165AB3DcFBaE0FC45F3c0685db4ec52ff730e

   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.04193478 ETH


Summary
=======
> Total deployments:   4
> Final cost:          0.04500914 ETH
```

versions:
@openzeppelin/truffle-upgrades: 1.7.0
Truffle: v5.3.6 (core: 5.3.6)
Node: v14.15.5
Ganache CLI: v6.12.2 (ganache-core: 2.13.2)

**Comment by frangio** [comment]
Thanks for reporting and for the easy reproduction steps. I've identified the issue, I'll let you know when it's fixed.

**Comment by frangio** [comment]
Fixed in https://github.com/OpenZeppelin/openzeppelin-upgrades/commit/bae728f15d7a30c7a07421681e50c4a467d57200 and published in `@openzeppelin/upgrades-core@1.7.4`!
